internal/admin/handle_authorized_apps.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package admin
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/exposure-notifications-server/internal/authorizedapp/database"
+	"github.com/google/exposure-notifications-server/internal/authorizedapp/model"
+	verdb "github.com/google/exposure-notifications-server/internal/verification/database"
+)
+
+// HandleAuthorizedAppsSave handles the create/update actions for authorized
+// apps.
+func (s *Server) HandleAuthorizedAppsSave() func(c *gin.Context) {
+	return func(c *gin.Context) {
+		var form authorizedAppFormData
+		err := c.Bind(&form)
+		if err != nil {
+			ErrorPage(c, err.Error())
+			return
+		}
+
+		ctx := c.Request.Context()
+		m := TemplateMap{}
+
+		aadb := database.New(s.env.Database())
+		if form.Action == "save" {
```

Comment:
Minor nit, it took reading through a bunch of code blocks to see that the only permitted actions were "save" and "delete". Perhaps we could restructure this into a switch so
```go
switch form.Action {
default:
     ErrorPage(c, fmt.Sprintf("Invalid request, app to edit not found, with action: %q", form.Action))

case "delete":
      ...

case "save":
      ...
}
```

Reply:
This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.